test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/DotnetProcessManager.cs

```diff
@@ -9,40 +9,51 @@
 
 namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
 {
-    internal static class MSBuildProcessManager
+    internal static class DotnetProcessManager
     {
-        public static Task<MSBuildResult> RunProcessAsync(
+        public static Task<ProcessResult> RunProcessAsync(
             ProjectDirectory project,
             string arguments,
             TimeSpan? timeout = null,
-            MSBuildProcessKind msBuildProcessKind = MSBuildProcessKind.Dotnet)
+            DotnetProcessKind dotnetProcessKind = DotnetProcessKind.Dotnet)
         {
             timeout = timeout ?? TimeSpan.FromSeconds(60);
 
+            var workingDirectory = project?.DirectoryPath;
+            if (string.IsNullOrEmpty(workingDirectory))
+            {
+                workingDirectory = Environment.CurrentDirectory;
+            }
+
             var processStartInfo = new ProcessStartInfo()
             {
-                WorkingDirectory = project.DirectoryPath,
+                WorkingDirectory = workingDirectory,
                 UseShellExecute = false,
                 RedirectStandardError = true,
                 RedirectStandardOutput = true,
             };
 
-            if (msBuildProcessKind == MSBuildProcessKind.Desktop)
+            if (dotnetProcessKind == DotnetProcessKind.DesktopMsBuild)
             {
                 if (string.IsNullOrEmpty(BuildVariables.MSBuildPath))
                 {
                     throw new ArgumentException("Unable to locate MSBuild.exe to run desktop tests. " +
-                        "MSBuild.exe is located using state created as part of running build[cmd|sh] at the root of the repository. Run build /t:Prepeare to set this up if this hasn't been done.");
+                        "MSBuild.exe is located using state created as part of running build[cmd|sh] at the root of the repository. Run build /t:Prepare to set this up if this hasn't been done.");
                 }
 
                 processStartInfo.FileName = BuildVariables.MSBuildPath;
                 processStartInfo.Arguments = arguments;
             }
-            else
+            else if (dotnetProcessKind == DotnetProcessKind.DotnetMsBuild)
             {
                 processStartInfo.FileName = "dotnet";
                 processStartInfo.Arguments = $"msbuild {arguments}";
             }
+            else
+            {
+                processStartInfo.FileName = "dotnet";
+                processStartInfo.Arguments = arguments;
+            }
```

Comment:
Can we separate all of this from the code path that runs MSBuild? Are you sure that running the shutdown command has the same requirments as the MSBuild junk?

Reply:
It doesn't need the project directory but it uses the same code to create a process and all that stuff. I thought reusing these made sense

Reply:
What parts does it need? I'd like to see this refactored in a more thoughtful way.

Reply:
> What parts does it need?

Everything in the method past this line. I'll try to refactor it better.